topics/data-science/tutorials/python-iterables/tutorial.md

```diff
@@ -0,0 +1,848 @@
+> __________
+> print(variants) # Should print {'B.1.1.7': 26267, 'B.1.351': 439, 'P.1': 384, 'B.1.617.2': 43486}
+> # Maybe we've exterminated B.1.1.7 and B.1.351, remove their numbers.
+> del _______
+> del _______
+> print(variants[______]) # Should print 384
+> print(variants[______]) # Should print 43486
+> ```
+>
+> > ### {% icon solution %} Solution
+> >
+> > variants = {
+> >   'B.1.1.7': 26267,
+> >   'B.1.351': 439,
+> > }
+> > variants['P.1'] = 384
+> > print(variants) # Should print {'B.1.1.7': 26267, 'B.1.351': 439, 'P.1': 384}
+> > variants['B.1.617.2'] = 43486
+> > print(variants) # Should print {'B.1.1.7': 26267, 'B.1.351': 439, 'P.1': 384, 'B.1.617.2': 43486}
+> > # Maybe we've exterminated B.1.1.7 and B.1.351, remove their numbers.
+> > del variants['B.1.1.7']
+> > del variants['B.1.351']
+> > print(variants['P.1']) # Should print 384
+> > print(variants['B.1.617.2']) # Should print 43486
+> > ```
+> >
+> {: .solution}
+{: .question}
+
+```python
+# Test code here!
+variants = {
+  'B.1.1.7': 26267,
+  'B.1.351': 439,
+}
+variants[_____] =  _____
+print(variants) # Should print {'B.1.1.7': 26267, 'B.1.351': 439, 'P.1': 384}
+__________
+print(variants) # Should print {'B.1.1.7': 26267, 'B.1.351': 439, 'P.1': 384, 'B.1.617.2': 43486}
+# Maybe we've exterminated B.1.1.7 and B.1.351, remove their numbers.
+del _______
+del _______
+print(variants[______]) # Should print 384
+print(variants[______]) # Should print 43486
+```
+
+
+# Choosing the Right Data Type
+
+Choosing the correct data type can sometimes require some thought, and even discussion with colleagues. And don't be afraid to search the internet for how other people have done it!
+
+
+Data type       | Examples                  | When to use it | When **not** to use it
+---             | ---                       | ---- | ----
+Boolean (`bool`) | `True`, `False` | If there are only two possible states, true or false | If your data is not binary
+Integer (`int`) | 1, 0, -1023, 42           | Countable, singular items. How many patients are there, how many events did you record, how many variants are there in the sequence | If doubling or halving the value would not make sense: do not use for e.g. patient IDs, or phone numbers. If these are integers you might accidentally do math on the value.
+Float (`float`) | 123.49, 3.14159, -3.33334 | If you need more precision or partial values. Recording distance between places, height, mass, etc.
+Strings (`str`) | 'patient_12312', 'Jane Doe', '火锅' | To store free text, identifiers, sequence IDs, etc. | If it's truly a numeric value you can do calculations with, like adding or subtracting or doing statistics.
+List / Array (`list`) | `['A', 1, 3.4, ['Nested']]` | If you need to store a list of items, like sequences from a file. Especially if you're reading in a table of data from a file. | If you want to retrieve individual values, and there are clear identifiers it might be better as a dict.
```

Comment:
i,m not sure if it is too soon or something for later. but perhaps should we let students play a bit more with nested lists and dictionaries

Reply:
yeah it can definitely be too soon. We can remove this.